archive/100-theorems-list/83_friendship_graphs.lean

```diff
@@ -124,10 +115,10 @@ end
   it is regular. We have shown that nonadjacent vertices of a friendship graph have the same degree,
   and if there isn't a politician, we can show this for adjacent vertices by finding a vertex
   neither is adjacent to, and then using transitivity. -/
-theorem is_regular_of_not_exists_politician (hG' : ¬exists_politician G) :
+theorem is_regular_of_not_exists_politician [nonempty V] (hG' : ¬exists_politician G) :
   ∃ (d : ℕ), G.is_regular_of_degree d :=
 begin
-  have v := arbitrary V,
+  have v : V := classical.choice infer_instance,
```

Comment:
I think you can use the tactic `inhabit V` or similar instead of invoking choice, which isn't necessary here.

Reply:
```suggestion
  inhabit V,
  have v := arbitrary V,
```

Reply:
The `by_cases` a couple of lines later invokes `classical.choice` anyway, but your comment led me to `classical.arbitrary`, which is better that what I'd written.

It might be nice to extend `inhabit` to have an additional form `inhabit V with v`, which would automatically write `have v := arbitrary V`.  Even better would be if it could do destructuring like `rcases`, so maybe it instead it adds `rcases arbitrary V with v`.